server/chat-plugins/othermetas.ts

```diff
@@ -187,91 +187,116 @@ export const commands: ChatCommands = {
-		} else {
-			buf += `<span class="col pokemonnamecol" style="white-space:nowrap"><a href="https://${Config.routes.dex}/items/${stone.id}" target="_blank">${stone.name}</a></span> `;
-		}
-		if (deltas.type && deltas.type !== 'mono') {
-			buf += `<span class="col typecol"><img src="https://${Config.routes.client}/sprites/types/${deltas.type}.png" alt="${deltas.type}" height="14" width="32"></span> `;
-		} else {
-			buf += `<span class="col typecol"></span>`;
-		}
-		buf += `<span style="float:left;min-height:26px">`;
-		buf += `<span class="col abilitycol">${megaSpecies.abilities['0']}</span>`;
-		buf += `<span class="col abilitycol"></span>`;
-		buf += `</span>`;
-		buf += `<span style="float:left;min-height:26px">`;
-		buf += `<span class="col statcol"><em>HP</em><br />0</span> `;
-		buf += `<span class="col statcol"><em>Atk</em><br />${deltas.baseStats.atk}</span> `;
-		buf += `<span class="col statcol"><em>Def</em><br />${deltas.baseStats.def}</span> `;
-		buf += `<span class="col statcol"><em>SpA</em><br />${deltas.baseStats.spa}</span> `;
-		buf += `<span class="col statcol"><em>SpD</em><br />${deltas.baseStats.spd}</span> `;
-		buf += `<span class="col statcol"><em>Spe</em><br />${deltas.baseStats.spe}</span> `;
-		buf += `<span class="col bstcol"><em>BST<br />${deltas.bst}</em></span> `;
-		buf += `</span>`;
-		buf += `</li>`;
-		this.sendReply(`|raw|<div class="message"><ul class="utilichart">${buf}<li style="clear:both"></li></ul></div>`);
-		this.sendReply(`|raw|<font size="1"><font color="#686868">Gen:</font> ${details["Gen"]}&nbsp;|&ThickSpace;<font color="#686868">Weight:</font> ${details["Weight"]}</font>`);
+		if (stone && dex.gen >= 8 && ['redorb', 'blueorb'].includes(stone.id)) {
+			return this.errorReply("The Orbs do not exist in Gen 8 and later.");
+		}
+		let stones;
+		if (!stone) {
+			const species = dex.getSpecies(targetid.replace(/(?:mega[xy]?|primal)$/, ''));
+			if (!species.exists) return this.errorReply(`Error: Mega Stone not found.`);
+			if (!species.otherFormes) return this.errorReply(`Error: Mega Evolution not found.`);
+			const megas = species.otherFormes.filter(poke => /(?:-Primal|-Mega(?:-[XY])?)$/.test(poke));
+			if (!megas.length) return this.errorReply(`Error: Mega Evolution not found.`);
+			stones = megas.map(poke => {
+				const megaPoke = dex.getSpecies(poke);
+				const flag = megaPoke.requiredMove === 'Dragon Ascent' ? megaPoke.requiredMove : megaPoke.requiredItem;
+				if (/mega[xy]$/.test(targetid) && toID(megaPoke.name) !== toID(dex.getSpecies(targetid))) return null;
+				if (!flag) {
+					console.log(megaPoke);
+					console.log(flag);
+					return null;
+				}
+				return getMegaStone(flag, sep[1]);
+			}).filter(poke => poke !== null);
+			if (!stones.length) return this.errorReply(`Error: Mega Evolution not found.`);
+		}
+		const toDisplay = (stones || [stone]), banlist = Dex.getFormat('gen8mixandmega').banlist;
+		toDisplay.forEach(aStone => {
```

Comment:
As per [the contributing guidelines](https://github.com/smogon/pokemon-showdown/blob/master/CONTRIBUTING.md#es5-and-es6), this should be a `for-of` loop instead.